Sources/XcodeGenKit/SourceGenerator.swift

```diff
@@ -9,6 +9,7 @@ struct SourceFile {
     let fileReference: PBXFileElement
     let buildFile: PBXBuildFile
     let buildPhase: BuildPhaseSpec?
+    var synchronizedRootGroup: PBXFileSystemSynchronizedRootGroup?
```

Comment:
`PBXFileSystemSynchronizedRootGroup` is a subclass of `PBXFileElement`, so maybe we should not introduce new var here?

Just change addition to `PBXProjGenerator`
from `let synchronizedRootGroups = sourceFiles.compactMap { $0.synchronizedRootGroup }` to
`let synchronizedRootGroups = sourceFiles.compactMap { $0. fileReference as? PBXFileSystemSynchronizedRootGroup }`

Reply:
https://github.com/yonaskolb/XcodeGen/pull/1557